src/inet/java/TcpSocketPeer.java

```diff
@@ -81,7 +81,10 @@ public TcpSocketPeer(Socket socket)
 // TLS
 //////////////////////////////////////////////////////////////////////////
 
-  public TcpSocket upgradeTls(TcpSocket self) { return upgradeTls(self, self); }
+  public TcpSocket upgradeTls(TcpSocket self, IpAddr addr, Long port)
```

Comment:
A few issues here I think:

There is no java method that matches `upgradeTls(self, self, addr, port)` so it doesn't compile.

Also, you need all the version of the java signatue matching the optional arguments:

```
TcpSocket upgradeTls(TcpSocket self) { return upgradeTls(self,null,null); }
TcpSocket upgradeTls(TcpSocket self, IpAddr addr) { return upgradeTls(self, addr, null); }
TcpSocket upgradeTls(TcpSocket self, IpAddr addr, Long port) {
   return upgradeTls(self, self, addr, port);  // if this is correct to do?
}
```